mongo/dataset_store.go

```diff
@@ -90,27 +90,12 @@ func (m *Mongo) GetDatasets(ctx context.Context, offset, limit int, authorised b
 	if err != nil {
 		log.Event(ctx, "error counting items", log.ERROR, log.Error(err))
 		if err == mgo.ErrNotFound {
-			return &models.DatasetUpdateResults{
-				Items:      []models.DatasetUpdate{},
-				Count:      0,
-				TotalCount: 0,
-				Offset:     offset,
-				Limit:      limit,
-			}, nil
+			return []*models.DatasetUpdate{}, totalCount, nil
 		}
-		return nil, err
+		return nil, 0, err
 	}
 
-	iter := q.Sort().Skip(offset).Limit(limit).Iter()
-
-	defer func() {
-		err := iter.Close()
-		if err != nil {
-			log.Event(ctx, "error closing iterator", log.ERROR, log.Error(err))
-		}
-	}()
```

Comment:
It looks like the functionality may have been changed here by removing this code. I think the original intention would have been to return all results of the limit is zero. Now the code is removed, it would return an empty set of results. Has this change been checked/confirmed?

Reply:
we spent some time trying to figure this out, what found is that this iter is never used, if the limit is 0 then an empty paginated list gets returned. We have component tests running on this test case and so can confirm this is an unused duplicate DB call, and that the functionality did not change from us removing it - happy to dicuss this though

Reply:
ah yes - I see that `iter` is not used in this case 😄👍